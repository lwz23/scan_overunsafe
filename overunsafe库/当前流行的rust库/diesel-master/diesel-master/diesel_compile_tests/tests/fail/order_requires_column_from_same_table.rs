extern crate diesel;

use diesel::*;

table! {
    users {
        id -> Integer,
    }
}

table! {
    posts {
        id -> Integer,
    }
}

allow_tables_to_appear_in_same_query!(users, posts);

fn main() {
    let source = users::table.order(posts::id);
}
