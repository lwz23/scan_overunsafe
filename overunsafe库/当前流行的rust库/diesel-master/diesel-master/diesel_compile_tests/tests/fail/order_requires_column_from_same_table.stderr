error[E0271]: type mismatch resolving `<table as AppearsInFromClause<table>>::Count == Once`
  --> tests/fail/order_requires_column_from_same_table.rs:20:31
   |
20 |     let source = users::table.order(posts::id);
   |                               ^^^^^ expected `Never`, found `Once`
   |
note: required for `posts::columns::id` to implement `AppearsOnTable<users::table>`
  --> tests/fail/order_requires_column_from_same_table.rs:13:9
   |
13 |         id -> Integer,
   |         ^^
   = note: associated types for the current `impl` cannot be restricted in `where` clauses
   = note: required for `SelectStatement<FromClause<users::table>>` to implement `OrderDsl<posts::columns::id>`
